sign/dilithium/eddilithium3.go

```diff
@@ -0,0 +1,96 @@
+// Code generated from mode.templ.go. DO NOT EDIT.
+
+package dilithium
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/cloudflare/circl/sign/dilithium/internal/common"
+	"github.com/cloudflare/circl/sign/eddilithium3"
+)
+
+// implEdMode3 implements the mode.Mode interface for EdDilithium3.
+type implEdMode3 struct{}
```

Comment:
nit: not the biggest fan of `impl` in type names. In a case like this I would usually just call this `edMode3`

Reply:
It's because of the type `implMode3` in `sign/dilithium` (which is generated from the same file) which then would be `mode3` which conflicts with the subpackage `sign/dilithium/mode3`.

Reply:
Got it. Yeah not a huge deal, I just don't like my Go looking like Java 😝

Reply:
Yeah, I don't like it either, but at least it doesn't show up in the API :).